File: 0049-group-anagrams/0049-group-anagrams.go
import (
    "sort"
)

func groupAnagrams(strs []string) [][]string {
    groups := make(map[string][]string)
    
    for _, word := range strs {
        sortedWord := sortString(word)
        groups[sortedWord] = append(groups[sortedWord], word)
        
    }
    
    result := make([][] string, 0, len(groups))
    for _, group := range groups {
        result = append(result, group)
    }
    
    
    return result
    
}

func sortString(s string) string {
    chars := []rune(s)
    
    sort.Slice(chars, func(i, j int) bool {
        return chars[i] < chars[j]
    })
    
    return string(chars)
}

File: 0049-group-anagrams/0049-group-anagrams.rs
use std::collections::HashMap;

impl Solution {
    pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
        let mut groups: HashMap<Vec<char>, Vec<String>> = HashMap::new();
        
        for word in strs {
            let mut chars: Vec<char> = word.chars().collect();
            chars.sort();
            groups.entry(chars).or_insert(Vec::new()).push(word);
        }
        
        groups.values().cloned().collect()
    }
}